test/commands/partial/pull.test.ts

```diff
@@ -0,0 +1,152 @@
+import * as path from "node:path";
```

Comment:
Are there any reader/writer dedicated tests we can replicate for partials? For examples:
knock-cli/test/lib/marshal/workflow/writer|reader.test.ts
knock-cli/test/lib/marshal/email-layout/writer|reader.test.ts

I know we test the inner read and writer code paths indirectly through these command tests, but I like setting up dedicated reader/writer tests for any granular behaviors and also to make it easier to write tests for future changes even if just basic ones.

Reply:
Sure, I'll set some up

Reply:
Actually, will do that in the pushing PR since that includes some reader updates, then I can cover all the tests at once